js/XAPI.js

```diff
@@ -265,10 +265,17 @@ class XAPI extends Backbone.Model {
 
     _.defer(() => {
       if (error) {
+        if (this.getConfig('_shouldRetryConnection') && !this.hasRetriedConnection) {
+          this.hasRetriedConnection = true;
```

Comment:
You sure you want only 1 reconnection attempt?

Reply:
Do you think a setting for retry attempt number instead of a boolean flag would be better? Default to 3 for example?

Reply:
As I don't know what is causing the connection failure, I can only imagine you needing more retries.